wirecard/oxid-ee
# WDPD-27: Transaction Table

### This PR

* Adds a new menu entry **Wirecard Transactions** to display a list of transactions
* Adds a searchable, paginated list that displays all transactions
* Adds a detail view to each transactions that is opened when clicking on a transaction

### Notes

* Unit tests are in progress
* The filter values for the *action* column currently don't match the entry values (e.g. "authorize-capture" vs. "authorization"). This is fixed in WDPD-24.

### How to Test

* Login to the admin
* Go to Extensions → Modules and activate *Wirecard Oxid EE Paymentgateway*
* Use a database manager to connect to the table `wdoxidee_ordertransactions` in the database
* Create one or more entries there; make sure to use a valid `OXID` order id for the field `WDOXIDEE_ORDERID`
* Go to Wirecard Transactions → Transactions and check if the created transactions are displayed
* Confirm that it is possible to sort the table (click on the table headers – note that tables can only be sorted in ascending order in Oxid) and search individual columns
* Click on a transaction to open the detail view(s)
* Navigate through the detail view tabs and check if the displayed information is correct
* In the *Details* tab, click the button *Copy XML* and check if it the text is copied into the clipboard

### Jira Links

* WDPD-27: https://jira.parkside.at/browse/WDPD-27
